lib/modem_key_mgmt/modem_key_mgmt.c

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <at_cmd.h>
+#include <at_cmd_parser/at_cmd_parser.h>
+#include <at_cmd_parser/at_params.h>
+#include <bsd_limits.h>
+#include <modem_key_mgmt.h>
+#include <stdio.h>
+#include <string.h>
+#include <zephyr.h>
+
+#define MODEM_KEY_MGMT_OP_LS "AT%CMNG=1"
+#define MODEM_KEY_MGMT_OP_RD "AT%CMNG=2"
+#define MODEM_KEY_MGMT_OP_WR "AT%CMNG=0"
+#define MODEM_KEY_MGMT_OP_RM "AT%CMNG=3"
+
+#define AT_CMNG_PARAMS_COUNT 5
+#define AT_CMNG_CONTENT_INDEX 4
+
+static char scratch_buf[4096];
+static char at_cmee_strings[2][10] = {"AT+CMEE=0", "AT+CMEE=1"};
+
+static int cmee_active(void)
+{
+	int result;
+	char response[sizeof("+CMEE: X\r\n") + 1];
+
+	memset(response, 0, sizeof(response));
+
+	result = at_cmd_write("AT+CMEE?", response, sizeof(response), NULL);
+	if ((result == 0) && strlen(response) > 0) {
+		if (strchr(&response[6], '1')) {
+			return 1;
+		} else {
+			return 0;
+		}
+	}
+
+	return -EIO;
+}
+
+static int cmee_set(bool enable)
+{
+	return at_cmd_write(at_cmee_strings[(int)enable], NULL, 0, NULL);
+}
+
+static int translate_error(int err)
+{
+	switch (err) {
+	case 513:
+		return -ENOENT;
+	case 514:
+		return -EPERM;
+	case 515:
+		return -ENOMEM;
+	case 518:
+		return -EACCES;
+	default:
+		return -EFAULT;
+	}
+}
+
+static int write_at_cmd_with_cme_enabled(char *cmd, char *buf, size_t buf_len,
+					 enum at_cmd_state *state)
+{
+	int retval;
+	int cmee_was_active = cmee_active();
+
+	if (cmee_was_active < 0) {
+		return -EFAULT;
+	}
+
+	if (!cmee_was_active) {
+		if (cmee_set(true)) {
```

Comment:
I would avoid writing statements inside `if` conditions and stick to the pattern:
```
err = foo()
if (err) {
}
...
```

Reply:
Why? We don't verify the actual value of the error code, we just check if error occured, current form is more concise. 
Personally, I would check for `cmee_set(true) != 0`, but I don't find that critical to have requested a change.

Reply:
I don't have a strong opinion here, but it is harder for me to read statements inside `if` conditions.
I don't expect `if` conditions to have side-effects.. if I am not interested in it I should just be able to jump over it. If there is a statement in the condition I'll have to check what it does instead.